docs/develop/dotnet/failure-detection.mdx

```diff
@@ -0,0 +1,55 @@
+---
+id: failure-detection
+title: Failure Detection .NET SDK feature guide
+sidebar_label: Failure Detection
+description: The Failure Detection section of the Temporal Developer's guide covers the many ways to detect failure with your Temporal application.
+toc_max_heading_level: 4
+keywords:
+  - guide context
+  - how to
+  - dotnet
+  - sdk
+  - failure detection
+tags:
+  - guide-context
+  - how-to
+  - dotnet
+  - sdk
+  - failure-detection
+---
+
+## Failure Detection {#failure-detection}
+
+## Workflow timeouts {#workflow-timeouts}
+
+Each Workflow timeout controls the maximum duration of a different aspect of a Workflow Execution.
+
+Workflow timeouts are set when [starting the Workflow Execution](#workflow-timeouts).
+
+- **[Workflow Execution Timeout](/workflows#workflow-execution-timeout)** - restricts the maximum amount of time that a single Workflow Execution can be executed.
+- **[Workflow Run Timeout](/workflows#workflow-run-timeout):** restricts the maximum amount of time that a single Workflow Run can last.
+- **[Workflow Task Timeout](/workflows#workflow-task-timeout):** restricts the maximum amount of time that a Worker can execute a Workflow Task.
+
+<!-- CONTENT-->
+
+### Workflow retries {#workflow-retries}
+
+A Retry Policy can work in cooperation with the timeouts to provide fine controls to optimize the execution experience.
+
+<!-- CONTENT-->
+
+## How to set Activity timeouts {#activity-timeouts}
```

Comment:
Inconsistent headings with workflow timeouts (this is prefixed with "How to set" but "Workflow timeouts" section is not)